test/e2e/framework/addon/vault/vault.go

```diff
@@ -156,15 +171,24 @@ func (v *Vault) Setup(cfg *config.Config, leaderData ...internal.AddonTransferab
 			},
 			{
 				Key: "server.standalone.config",
-				Value: `
+				Value: fmt.Sprintf(`
 				listener "tcp" {
-					tls_disable = false
 					address = "[::]:8200"
 					cluster_address = "[::]:8201"
 					tls_disable = false
+					tls_client_ca_file = "/vault/tls/ca.crt"
 					tls_cert_file = "/vault/tls/server.crt"
 					tls_key_file = "/vault/tls/server.key"
-				}`,
+					tls_require_and_verify_client_cert = %s
+				}`, strconv.FormatBool(v.EnforceMtls)),
+			},
+			{
+				Key:   "server.extraEnvironmentVars.VAULT_CLIENT_CERT",
+				Value: "/vault/tls/server.crt",
+			},
+			{
+				Key:   "server.extraEnvironmentVars.VAULT_CLIENT_KEY",
+				Value: "/vault/tls/server.key",
 			},
```

Comment:
This perhaps looks weird, but it is essentially correct, that Vault server certificate was issued with the same CA which is now configured under the `tls_client_ca_file` configuration of the TCP listener, so they are completely valid client certificate as well, those environment variables are needed for the `exec` command in the Kubernetes Liveness and Readiness probe to work, the Probes are using the Vault CLI to check the status of the Vault server and therefore require a valid client certificate to interact with the status API at server side.

Reply:
Would it be possible you generate a new keypair instead of regenerating the existing pair?

Reply:
Sure, I am actually already generating a new certificate specifically for the clients in the e2e framework and storing it in a Kubernetes secret, I will then mount this secret in the Vault pod and use the client related certificate in those environment variables instead of the server certificate.

Reply:
@inteon and @wallrj, the configuration above was updated to use the client certificate in the environment variable, could you please have another look?